src/coreclr/hosts/corerun/corerun.cpp

```
@@ -498,6 +515,16 @@ bool TryRun(const int argc, const wchar_t* argv[], Logger &log, const bool verbo
     nativeDllSearchDirs.Append(W(";"));
     nativeDllSearchDirs.Append(hostEnvironment.m_coreCLRDirectoryPath);
 
+    // Preload mock hostpolicy if requested.
+    StackSString hostpolicyPath;
+    if (WszGetEnvironmentVariable(W("MOCK_HOSTPOLICY"), hostpolicyPath) > 0 && hostpolicyPath.GetCount() > 0)
+    {
+        if (!TryLoadHostPolicy(hostpolicyPath))
+        {
+            log << W("Unable to load requested mock hostpolicy.");
```

Why is this okay? Tests should fail hard if they don't operate as expected.